lib/randomfileaccess.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+"use strict";
+
+const PRIVATE = Symbol("PRIVATE");
+const { open } = require("fs").promises;
+
+/**
+ * @typedef {Object} WriteResponse
+ * @property {Number} bytesWritten the number of bytes written
+ * @property {Buffer} buffer the buffer written
+ */
+
+/**
+ * @typedef {Object} SubRange
+ * @property {Number} low Low-end of the range
+ * @property {Number} high High-end of the range (included)
+ * @property {Number} length Length of the range
+ */
+
+/**
+ * Open a file handle in the given mode, grow the file up to the expected target size
+ * 
+ * @param {String|URL} path File path
+ * @param {"r"|"w"} flags Mode to open the file
+ * @param {Number} [size] expected size of the file
+ * @returns {FileHandle} File handle
+ */
+async function openFile(path, flags, size) {
+    const handle = await open(path, flags);
+    try {
+        if ((flags === "w") && size) {
```

Comment:
What if someone would try to use `rw`? Please check that, and log it if it's the case (or throw).

Reply:
That would only be possible if you change the RandomFileAccess class. We only pass "r" or "w" from the class.